### PR TITLE
SWPWA-561 - Desktop menu redesign

### PR DESCRIPTION
**Pull request by AivarsA**
- MenuOverlay has been renamed to Menu (since it is not an overlay anymore)
- Horizontal menu for desktop, mobile stays the same

Third level menu types -
- Default - just title (no class needed)

- Logo - Title with a small image next to it - add `Menu-ItemFigure_type_logo` class to parent item (second level item) when creating third-level items in backend and add an image to third-level items

- Banner - Large image with a button - add `Menu-ItemFigure_type_banner` class to parent item (second level item) when creating third-level items in backend and add an image to third-level items

### CODE REVIEW

File: src/app/component/Menu/Menu.component.js [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    renderStoreSwitcher() {
+        return <StoreSwitcher />;
+    }
+
+    render() {
+        return (
+            <div
+              block="Menu"
+              elem="MenuWrapper"
+              onMouseLeave={ () => this.closeMenu() }
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.component.js:417]: The usage of `() => this.closeMenu()` is redundant. You can directly use `this.closeMenu`. This is related to every call of `() => this.closeMenu()`.

File: src/app/component/Menu/Menu.component.js [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+        const { menu } = this.props;
+        const categoryArray = Object.values(menu);
+
+        if (!categoryArray.length) {
+            return null;
+        }
+
+        const {
+            0: { children, title: mainCategoriesTitle }
+        } = categoryArray;
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.component.js:386]: Maybe we can use this instead?

```js
const [{ children, title: mainCategoriesTitle }] = categoryArray;
```

File: src/app/component/Menu/Menu.style.scss [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+            }
+
+            &:hover,
+            &:focus {
+                @include desktop {
+                    color: var(--menu-item-hover-color);
+                    text-decoration: none;
+                }
+            }
+        }
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.style.scss:311]: Can we escape this nesting?

File: src/app/component/Menu/Menu.style.scss [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+
+        .Menu-Link {
+            color: var(--menu-third-level-color);
+            font-size: 11px;
+            margin-top: 20px;
+            text-transform: none;
+        }
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.style.scss:335]: Or this?

File: src/app/component/Menu/Menu.style.scss [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+            padding: 0 55px 40px;
+        }
+
+        .Menu-Link {
+            @include desktop {
+                margin-left: 0;
+                margin-right: 55px;
+            }
+        }
+    }
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.style.scss:350]: I think these are unused.

File: src/app/component/Menu/Menu.style.scss [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+        .Menu-Link {
+            padding-left: 0;
+
+            &:hover,
+            &:focus {
+                @include desktop {
+                    color: var(--menu-item-hover-color);
+                }
+            }
+        }
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.style.scss:370]: Again this one, hmm

File: src/app/component/Menu/Menu.style.scss [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+
+        li {
+            width: 1.9rem;
+
+            @include desktop {
+                margin-bottom: 0;
+                width: 16px;
+            }
+        }
+    }
```
Comment (alfredsgenkins) [inline src/app/component/Menu/Menu.style.scss:415]: I think we should move this to CMS block styles. There, we allow styling like this, by selector.

**Comment by alfredsgenkins** [comment]
So, I added multiple comments. I could of add more, but most of them are based around following bullet-points:
- No conditional rendering ?: and && in JSX
- Can we prevent using “/* eslint-disable react/jsx-no-bind */”
- I would like to prevent “{ childrenArray.map((item) => {}) }” from being used anywhere
- Prevent unnecessary brackets from usage: “() => this.closeMenu()”
- Prevent passing tons of arguments around, use props instead and enforce new components

Read the main comments about QA as well.

File: src/app/component/Menu/Menu.component.js [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+
+    renderItemContentImage(icon, itemMods) {
+        if (!icon) {
+            return null;
+        }
+
+        return (
+            <Image
+              mix={ { block: 'Menu', elem: 'Image', mods: itemMods } }
+              src={ icon && media(icon) }
```
[the remark this answers is not in the document]
Comment (AivarsA) [reply src/app/component/Menu/Menu.component.js:95]: This was done already in MenuOverlay component, it was already there

**Pull request merged**